server/server.go

```diff
@@ -2535,7 +2535,7 @@ func (s *Server) setFirstPingTimer(c *client) {
 		if d > firstPingInterval {
 			d = firstPingInterval
 		}
-	} else if d > firstClientPingInterval {
+	} else if d > firstClientPingInterval && !opts.DisableShortFirstPing {
```

Comment:
We should update the comment of this function since it now applies to all connection types. Also, I am not sure we really need a different interval for various type of connections.
Also, I propose that we update code to set the initial RTT for all type of connections on processing of CONNECT and for LEAF, GW and ROUTE also do that on processing of first INFO (since when they solicit they send CONNECT+INFO, but only receive INFO from peer - we also need to ensure that 'start' is set in create of all type of connections, which may not be the case). That would give all connections in the server an initial rtt. We can then ask ourselves if we even need the firstPingInterval or firstClientPingInterval at all. I think we could still keep that so that we get a "real" RTT, but again, not sure we need different values for client and other types.

Reply:
These are all really good points. I think that would be a separate PR. Will merge this one then we can decide on your suggestion. I like it though.